source/blog/2018-05-30-emberjs-accessibility-and-a11y-tools.md

```diff
@@ -0,0 +1,168 @@
+
+![Errors in the tests route](/images/blog/2018-05-30-emberjs-accessibility-and-a11y-tools/route-tests-errors.png)
+
+This is awesome because we don't need to become an expert on form accessibility to get started down the right track.
+
+### Fix the critical problems identified in the tests
+
+Since the errors have ratings, we know to start with the critical error, which is that a form has missing labels. Screen readers rely on labels, not placeholder text, to help users navigate forms. Also, it's better UX for everyone.
+
+Here's what the form looks like before changes:
+
+```handlebars
+{{input id="filterByCity" value=value key-up=(action 'handleFilterEntry') class="light" placeholder="Filter By City"}}
+```
+
+![A form without labels](/images/blog/2018-05-30-emberjs-accessibility-and-a11y-tools/form-without-labels.png)
+
+Let's add the label. The `for` attribute should match the input id.
+
+```handlebars
+<label for="filterByCity" class="screen-reader">
+    Filter By City
+</label>
+
+{{input id="filterByCity" value=value key-up=(action 'handleFilterEntry') class="light" placeholder="Filter By City"}}
+```
+
+![A form with labels](/images/blog/2018-05-30-emberjs-accessibility-and-a11y-tools/form-with-labels.png)
+
+Now the filter has a label, and the tests pass! 
+
+Fixing issues with contrast, aria labels, and DOM elements is an important step towards building an accessible app. That said, there are some tools that help reduce the amount of work that you have to do by hand, and we'll cover those next.
+
+## Using ember-a11y-landmarks for roles
+
+Here's just one example of an addon that helps reduce the number of changes to make by hand. The [ember-a11y-landmarks](https://github.com/ember-a11y/ember-a11y-landmarks) addon helps manage the roles that should go on semantic tags like `<header>` and `<nav>`, without needing to learn the intricacies of when to use them.
+
+One huge thing you can do to improve an app's accessibility is to use semantic tags in the HTML. For example, always use `<button>` for buttons, instead of making them `divs`. However, some semantic tags still need extra attributes called roles for screen readers to work right, and they only need them some of the time based on their position in the DOM. ember-a11y-landmarks to the rescue!
+
+To use this addon, wrap content in the ally-landmark block and say what kind of tag to use. Here's a template example:
+
+```handlebars
+{{#a11y-landmark tagName="footer"}}
+   This is a footer!
+{{/a11y-landmark}}
+```
+
+And here's the html output with the correct role assigned:
+
+```
+<footer id="ember337" role="contentinfo" class="ember-view"> 
+  This is a footer!
+</footer>
+```
+
+There's also some good news! Some of the use cases for this addon are addressed by new options available [in Ember itself}(#new-options-for-out-of-the-box-accessibility)!
```

Comment:
I think the `}` after `itself` ought be a `]` :)

Reply:
good catch! fixed